### PR TITLE
Support mark dynamic

### PR DESCRIPTION
**Pull request by yitongh**

### CODE REVIEW

File: torchacc/core/dynamic.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        x (torch.Tensor): input tensor
+        dims (Union[List[int], int]): the dim marked as dynamic, can be a single dim or
+            multiple dims
+        bounds (Union[List[int], int]): upper bounds corresponding to the dynamic dims
+    """
+    if ta.get_global_context().python_dispatcher is None:
+        # Some symint operators require this in order for the operator dispatch
+        # to work properly
+        ta.get_global_context().python_dispatcher = enable_python_dispatcher()
+    if isinstance(dims, int):
```
Comment (anw90) [inline torchacc/core/dynamic.py:23]: Add an assert to ensure that dims and bounds have the same type.

File: torchacc/core/dynamic.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        dims = [dims]
+        bounds = [bounds]
+    assert isinstance(dims, list), "dims should be of int or list type"
```
Comment (anw90) [inline torchacc/core/dynamic.py]: Replace "dims should be of int or list type" with "dims should be of type int or list of int"

File: torchacc/core/dynamic.py (hunk 1 above)
Comment (yitongh) [reply torchacc/core/dynamic.py:23]: done

File: torchacc/core/dynamic.py (hunk 2 above)
Comment (yitongh) [reply torchacc/core/dynamic.py]: done

**Review by anw90 — APPROVED** [review]

**Pull request merged**